### PR TITLE
Adding in access tier support

### PR DESCRIPTION
**Pull request by amnguye**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/tests/PageBlobClientTests.cs [hunk 1]
```diff
@@ -1380,6 +1380,122 @@ public async Task StartCopyIncrementalAsync_AccessConditionsFail()
...
+        }
+
+        [Test]
+        public async Task StartCopyIncrementalAsync_AccessTierFail()
+        {
+            var premiumService = this.GetServiceClient_PremiumBlobAccount_SharedKey();
+            using (this.GetNewContainer(out var container, service: premiumService, premium: true))
+            {
+                // Arrange
+                //await container.SetAccessPolicyAsync(PublicAccessType.Container);
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/tests/PageBlobClientTests.cs]: Why is this commented out?

File: sdk/storage/Azure.Storage.Blobs/tests/BlobClientTests.cs [hunk 2]
```diff
@@ -288,6 +288,99 @@ async Task Verify(Task<Response<BlobContentInfo>> upload)
...
+
+                    try
+                    {
+                        File.WriteAllBytes(path, data);
+
+                        var file = new FileInfo(path);
+
+                        var options = new ParallelTransferOptions { MaximumThreadCount = maximumThreadCount };
+
+                        await Verify(blob.UploadAsync(
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/tests/BlobClientTests.cs]: Nit - you probably don't need the `Verify` function if you're only doing one call.  This pattern is primarily for scenarios where we've got four or five values to test out.

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs [hunk 3]
```diff
@@ -1287,13 +1291,15 @@ static Task WritePartitionAsync(Response<BlobDownloadInfo> response, Stream dest
         public virtual Operation<long> StartCopyFromUri(
             Uri source,
             Metadata metadata = default,
+            AccessTierOptional? accessTier = default,
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs]: We don't want a type named `AccessTierOptional` - it doesn't match the naming conventions for .NET types.  What is this type and how does it differ from `AccessTier`?

Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs]: Apparently this is a remnant of a workaround for the swagger being in a partially updated state.  You can fix this by:

- Deleting the section https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/swagger/readme.md#make-accesstier-unique
- Pulling the latest changes from master
- Regenerating according to the instructions at https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/CONTRIBUTING.md#code-generation
- `AccessTierOptional` will just become `AccessTier` and you can replace it anywhere the compiler complains

File: sdk/storage/Azure.Storage.Blobs/tests/BlockBlobClientTests.cs [hunk 4]
```diff
@@ -898,6 +898,93 @@ public async Task CommitBlockListAsync_Error()
...
+                // Assert
+                var blobList = await blob.GetBlockListAsync(BlockListType.All);
+                Assert.AreEqual(2, blobList.Value.CommittedBlocks.Count());
+                Assert.AreEqual(this.ToBase64(firstBlockName), blobList.Value.CommittedBlocks.First().Name);
+                Assert.AreEqual(this.ToBase64(secondBlockName), blobList.Value.CommittedBlocks.ElementAt(1).Name);
+                Assert.AreEqual(1, blobList.Value.UncommittedBlocks.Count());
+                Assert.AreEqual(this.ToBase64(thirdBlockName), blobList.Value.UncommittedBlocks.First().Name);
+
+                var response = await blob.GetPropertiesAsync();
+                Assert.AreEqual(AccessTierOptional.Cool.ToString(), response.Value.AccessTier);
```
[the remark this answers is not in the document]
Comment (amnguye) [reply sdk/storage/Azure.Storage.Blobs/tests/BlockBlobClientTests.cs]: It doesn't. It tries to compare Cool to "Cool" and doesn't accept it.

**Pull request closed without merging**